kommunicateui/src/main/java/com/applozic/mobicomkit/uiwidgets/conversation/fragment/MobiComConversationFragment.java

```diff
@@ -3913,20 +3919,44 @@ public void onAction(boolean didGrant) {
             }
         });
 
+        multiSelectGalleryButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                emoticonsFrameLayout.setVisibility(View.GONE);
+                if (getActivity() != null) {
+                    if (!((KmStoragePermissionListener) getActivity()).isPermissionGranted()) {
+                        //get permission
+                        ((KmStoragePermissionListener) getActivity()).checkPermission(new KmStoragePermission() {
+                            @Override
+                            public void onAction(boolean didGrant) {
+                                //did not get permission
+                                if (!didGrant) {
+                                    return;
+                                }
+                            }
+                        });
+                    } else {
+                        ((ConversationActivity) getActivity()).setFromMultiSelectGallery(true);
+                        ((ConversationActivity) getActivity()).processMultiSelectGallery();
```

Comment:
@shubhamtewari what will happen if someone directly call processMultiSelectGallery without setting setFromMultiSelectGallery?

Reply:
TL;DR the processMultiSelectGallery() wont be called
Why:
since there is no permission, the function statements will be skipped (internal working). after permission is granted, in the callback, it will be checked where the permission call came from (by checking the fromMultiSelectGallery). Since it is false, the processMultiSelectGallery wont be called again.

Reply:
there are other calls of processMultiSelectGallery() that do not require this to be true. thus it is set from outside.